### PR TITLE
Using NIL_P macro instead of `== Qnil`

### PR DESCRIPTION
**Pull request by S-H-GAMELINKS**
Using `NIL_P` macro instead of `== Qnil`, because to more simply code.

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -3466,7 +3466,7 @@ rb_ary_bsearch_index(VALUE ary)
 	    satisfied = 1;
 	    smaller = 1;
 	}
-	else if (v == Qfalse || v == Qnil) {
+	else if (v == Qfalse || NIL_P(v)) {
```
Comment (nobu) [inline array.c:3469]:
```suggestion
	else if (!RTEST(v)) {
```

File: iseq.c [hunk 2]
```diff
@@ -2413,7 +2413,7 @@ iseq_iterate_children(const rb_iseq_t *iseq, void (*iter_func)(const rb_iseq_t *
                 UNALIGNED_MEMBER_PTR(body->catch_table, entries[i]);
             child = entry->iseq;
             if (child) {
-                if (rb_hash_aref(all_children, (VALUE)child) == Qnil) {
+                if (rb_hash_aref(all_children, NIL_P((VALUE)child))) {
```
Comment (nobu) [inline iseq.c:2416]:
```suggestion
                if (NIL_P(rb_hash_aref(all_children, (VALUE)child))) {
```

File: iseq.c [hunk 3]
```diff
@@ -2432,7 +2432,7 @@ iseq_iterate_children(const rb_iseq_t *iseq, void (*iter_func)(const rb_iseq_t *
               case TS_ISEQ:
                 child = (const rb_iseq_t *)code[i+j+1];
                 if (child) {
-                    if (rb_hash_aref(all_children, (VALUE)child) == Qnil) {
+                    if (rb_hash_aref(all_children, NIL_P((VALUE)child))) {
```
Comment (nobu) [inline iseq.c:2435]:
```suggestion
                    if (NIL_P(rb_hash_aref(all_children, (VALUE)child))) {
```

File: range.c [hunk 4]
```diff
@@ -643,7 +643,7 @@ bsearch_integer_range(VALUE beg, VALUE end, int excl)
 	    satisfied = val; \
 	    smaller = 1; \
 	} \
-	else if (v == Qfalse || v == Qnil) { \
+	else if (v == Qfalse || NIL_P(v)) { \
```
Comment (nobu) [inline range.c:646]:
```suggestion
	else if (!RTEST(v)) { \
```

**Review by nobu — CHANGES_REQUESTED** [review]

**Comment by S-H-GAMELINKS** [comment]
@nobu 
Thanks for your reviewed.
Fixed it.

**Review by nobu — APPROVED** [review]

**Pull request merged**